python/tskit/tables.py

```diff
@@ -374,6 +374,43 @@ def __getitem__(self, index):
 
         return ret
 
+    def __setitem__(self, index, new_row):
+        """
+        Replaces a row of this table at the specified index with information from a
+        row-like object. Metadata, will be validated and encoded according to the table's
+        :attr:`metadata_schema<tskit.IndividualTable.metadata_schema>`.
+
+        :param index: the zero-index of the row to change
+        :param row-like new_row: An object that has attributes corresponding to the
+            properties of the new row. Both the objects returned from ``table[i]`` and
+            e.g. ``ts.individual(i)`` work for this purpose, along with any other
+            object with the correct attributes.
+        """
+        if isinstance(index, numbers.Integral):
+            # Single row by integer
+            if index < 0:
+                index += len(self)
+            if index < 0 or index >= len(self):
+                raise IndexError("Index out of bounds")
+        else:
```

Comment:
Let's just think ahead here to make sure we're not painting ourselves into a corner. What would the semantics be for other types of updates? I guess we could support other types of assignment here.

If the index argument is a slice/boolean array, then the ``new_row`` argument should be a Sequence of row_like values (of the same length as the index argument), and then we sequentially call ``update_row`` on these? Wouldn't be particularly efficient, but we could push the implementation down into C if we wanted to, and it would allow us to do things like

```python
tables.edges[:] = tables.edges[::-1]
```

which I'm sure would be handy. 

I'm not suggesting we do this now, just making sure we leave the door open for it in the future. I think TypeError on anything that's not an Integral for the index value does this?

Reply:
Yep, that all makes sense. The code as-is raises `TypeError` on anything non-`Integral`.